server/rest/src/main/java/org/infinispan/rest/resources/ResourceUtil.java

```diff
@@ -0,0 +1,59 @@
+package org.infinispan.rest.resources;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Util class for REST resources.
+ *
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+class ResourceUtil {
+   private final static Log logger = LogFactory.getLog(ResourceUtil.class, Log.class);
+
+   static CompletableFuture<RestResponse> notFoundRspFuture() {
+      return CompletableFuture.completedFuture(
+            new NettyRestResponse.Builder()
+                  .status(HttpResponseStatus.NOT_FOUND)
+                  .build()
+      );
+   }
+
+   static RestResponse objToJsonRsp(Object o, InvocationHelper invocationHelper) {
+      return addJsonToRspBuilder(o, new NettyRestResponse.Builder(), invocationHelper).build();
+   }
+
+   static CompletableFuture<RestResponse> objToJsonRspFuture(Object o, InvocationHelper invocationHelper) {
+      return completedFuture(objToJsonRsp(o, invocationHelper));
+   }
+
+   static CompletableFuture<RestResponse> objToJsonRspFuture(Object o, NettyRestResponse.Builder responseBuilder, InvocationHelper invocationHelper) {
+      RestResponse response = addJsonToRspBuilder(o, responseBuilder, invocationHelper).build();
+      return completedFuture(response);
+   }
+
+   static NettyRestResponse.Builder addJsonToRspBuilder(Object o, NettyRestResponse.Builder responseBuilder, InvocationHelper invocationHelper) {
+      try {
+         byte[] bytes = invocationHelper.getMapper().writeValueAsBytes(o);
+         responseBuilder.contentType(APPLICATION_JSON).entity(bytes).status(OK);
+      } catch (JsonProcessingException e) {
+         logger.error(e);
+         responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).entity(e.getMessage());
```

Comment:
I am not sure if @gustavonalle was trying to hide the cause of the server error or not before. I know it can sometimes be frowned upon exposing some stuff to clients.

Reply:
The idea is to expose the error in a structured way, like query is doing, in JSON with a message and the cause: https://github.com/infinispan/infinispan/blob/a922cc2104ead99670614cae441937d35e04a9ce/remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/json/JsonQueryErrorResult.java#L15

Looks like it is a good time to do it @ryanemerson

Reply:
Makes sense, I'll add a `JsonErrorResult` class.